libcontainer/configs/validate/validator.go

```diff
@@ -262,16 +253,72 @@ func cgroupsCheck(config *configs.Config) error {
 	return nil
 }
 
+func checkIDMapMounts(config *configs.Config, m *configs.Mount) error {
+	if !m.IsIDMapped() {
+		return nil
+	}
+
+	if !m.IsBind() {
+		return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is supported only for mounts with the option 'bind'", m)
+	}
+	if config.RootlessEUID {
+		return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is not supported when runc is being launched with EUID != 0, needs CAP_SYS_ADMIN on the runc parent's user namespace", m)
+	}
+	if len(config.UidMappings) == 0 || len(config.GidMappings) == 0 {
+		return fmt.Errorf("invalid mount %+v: not yet supported to use GidMappings/UidMappings in a mount without also using a user namespace", m)
+	}
+	if !sameMapping(config.UidMappings, m.UIDMappings) {
+		return fmt.Errorf("invalid mount %+v: not yet supported for the mount uidMappings to be different than user namespace uidMapping", m)
+	}
+	if !sameMapping(config.GidMappings, m.GIDMappings) {
+		return fmt.Errorf("invalid mount %+v: not yet supported for the mount gidMappings to be different than user namespace gidMapping", m)
+	}
+	if !filepath.IsAbs(m.Source) {
+		return fmt.Errorf("invalid mount %+v: mount source not absolute", m)
+	}
+
+	return nil
+}
+
 func mounts(config *configs.Config) error {
 	for _, m := range config.Mounts {
+		// We upgraded this to an error in runc 1.2. We might need to
+		// revert this change if some users haven't still moved to use
```

Comment:
As you say, we might need to revert this change. So let's double check.

Reply:
Let's unify the discussion in the other thread.